### PR TITLE
[imageio] Support for compilation with libheif 1.17

### PR DESCRIPTION
**Pull request by victoryforce**
We have to include extra header file starting with libheif 1.17.

### CODE REVIEW

**Comment by kmilos** [comment]
Hm, not sure about this... what requires it actually? Could it be a libheif bug? Normally the single public header _should_ be it?

File: src/imageio/imageio_heif.c [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with darktable.  If not, see <http://www.gnu.org/licenses/>.
+*/
 
 #include "common/image.h"
+
 #include <libheif/heif.h>
+#if LIBHEIF_HAVE_VERSION(1, 17, 0)
```
Comment (TurboGit) [inline src/imageio/imageio_heif.c:22]: Will that work for 1.17.1 and above?

**Comment by TurboGit** [comment]
> Hm, not sure about this... what requires it actually? Could it be a libheif bug? Normally the single public header _should_ be it?

Maybe related to:

https://github.com/darktable-org/darktable/actions/runs/6541838497/job/17763930944?pr=15437#step:4:954

File: src/imageio/imageio_heif.c (hunk 1 above)
Comment (victoryforce) [reply src/imageio/imageio_heif.c:22]: Yes, it will. This macro expands to `LIBHEIF_NUMERIC_VERSION >= LIBHEIF_MAKE_VERSION(1, 17, 0)`.

**Comment by victoryforce** [comment]
> Maybe related to:
> 
> https://github.com/darktable-org/darktable/actions/runs/6541838497/job/17763930944?pr=15437#step:4:955

Exactly.

> Could it be a libheif bug?

It is hardly possible to prevent you from considering Dirk Farin's clear decision to split the header as a bug. But that's what is a fact in libheif 1.17.0.

**Comment by kmilos** [comment]
> It is hardly possible to prevent you from considering Dirk Farin's clear decision to split the header as a bug. But that's what is a fact in libheif 1.17.0.

Ok, he clearly thought managing item properties as a different/advanced level of library use. I guess it makes sense now.

**Review by TurboGit — APPROVED** [review]
So merging, thanks!

**Pull request merged**